astroquery/esasky/core.py

```diff
@@ -107,9 +108,10 @@ def query(self, query, *, output_file=None, output_format="votable", verbose=Fal
         -------
         A table object
         """
-        job = self._tap.launch_job(query=query, output_file=output_file,
-                                   output_format=output_format,
-                                   verbose=verbose,
+        if not verbose:
+            commons.suppress_vo_warnings()
+            warnings.filterwarnings("ignore", category=astropy.units.core.UnitsWarning)
```

Comment:
This is modifying the warnings filter globally, it would be better to use the `warnings.catch_warnings()` context manager.

Reply:
Thanks for teaching me something new today! I've now added a context manger. Do you have any opinions regarding the suppression of specific warnings like done above or if it is more in line with the astroquery guidelines to use `warnings.simplefilter("ignore")` instead? @bsipocz

Reply:
Recently I did this very specific ignore for windows, and don't have a better suggestion for this case either. Make the ignore module or small code part specific, and be as specific as possible to not overcatch and ignore warnings:

https://github.com/astropy/astroquery/blob/main/astroquery/sdss/core.py#L940

As for units related issues, other modules are casting them over for a fix (e.g. nasa_exoplanet_archive), but I think that's less applicable for the `esasky` case as here you may run into much more, and less predictable unit issues (for them it's about one set of units the remote services uses that they need to cast into valid astropy units, while here it's some non standard compliancy of the votables in the response).

Reply:
Alright! Let's keep it as is then. It is unfeasable for ESASky to repair FITS file data, since ESASky is exposing data from so many missions.